### PR TITLE
Support for Elasticsearch HTTP Authentification

### PR DESCRIPTION
**Pull request by ycyr**
Hi Peter, thanks for your Prometheus Elasticsearch adapter. I've just added support for Elasticsearch HTTP Authentification.  I've also modified the docker-compose.yml file for testing the HTTP Authentification.

Regards,

Yanick

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ A read and write adapter for prometheus persistent storage
 | Env Variables     | Default               | Description                                                        |
 | ----------------- | --------------------- | ------------------------------------------------------------------ |
 | ES_URL            | http://localhost:9200 | Elasticsearch URL                                                  |
+| ES_USER           |                       | Elasticsearch User                                                 |
+| ES_PWD            |                       | Elasticsearch User Password                                        |
```
Comment (pwillie) [inline README.md]: nit: would prefer `ES_PASSWORD`

File: lib/elasticsearch/adapter.go [hunk 2]
```diff
@@ -37,6 +37,8 @@ type Adapter struct {
 	indexMaxAge   string
 	indexMaxDocs  int64
 	esURL         string
+	esUser        string
+        esPwd         string
```
Comment (pwillie) [inline lib/elasticsearch/adapter.go]: nit: whitespace issue and `esPassword`

File: main.go [hunk 3]
```diff
@@ -20,6 +20,8 @@ import (
 func main() {
 	var (
 		url           = flag.String("es_url", "http://localhost:9200", "Elasticsearch URL.")
+		user          = flag.String("es_user", "", "Elasticsearch User.")
+                pass          = flag.String("es_pwd", "", "Elasticsearch User Password.")
```
Comment (pwillie) [inline main.go]: nit: whitespace and `es_password`

File: main.go [hunk 4]
```diff
@@ -54,6 +56,8 @@ func main() {
 	elastic, err := elasticsearch.NewAdapter(
 		log,
 		elasticsearch.SetEsUrl(*url),
+		elasticsearch.SetEsUser(*user),
+                elasticsearch.SetEsPwd(*pass),
```
Comment (pwillie) [inline main.go]: nit: whitespace

File: README.md (hunk 1 above)
Comment (ycyr) [reply README.md]: Sound good

File: lib/elasticsearch/adapter.go (hunk 2 above)
Comment (ycyr) [reply lib/elasticsearch/adapter.go]: Typo

File: main.go (hunk 3 above)
Comment (ycyr) [reply main.go]: Typo

File: main.go (hunk 4 above)
Comment (ycyr) [reply main.go]: Typo

**Comment by ycyr** [comment]
Hi Peter, can you retake a look. I made the changed you requested.

Regards

**Pull request merged**